frontend/src/lib/components/Admin/CalendarEvents.svelte

```diff
@@ -0,0 +1,339 @@
+	currentCalendarEvent = calendarEvent;
+	if (calendarEvent) {
+		editForm.title = calendarEvent.title || "";
+		editForm.description = calendarEvent.description || "";
+		editForm.external_link = calendarEvent.external_link || "";
+		editForm.event_date = calendarEvent.event_date;
+	} else {
+		editForm.title = "";
+		editForm.description = "";
+		editForm.external_link = "";
+		editForm.event_date = "";
+	}
+	showEditModal = true;
+}
+
+function openDeleteModal(calendarEvent: CalendarEventRead) {
+	currentCalendarEvent = calendarEvent;
+	showDeleteModal = true;
+}
+
+async function saveCalendarEvent() {
+	try {
+		if (currentCalendarEvent) {
+			const updateData: CalendarEventUpdate = {
+				title: editForm.title || null,
+				description: editForm.description || null,
+				external_link: editForm.external_link || null,
+				event_date: editForm.event_date || null,
+			};
+
+			const { error: apiError } = await updateEvent({
+				path: { event_id: currentCalendarEvent.id },
+				body: updateData,
+			});
+
+			if (apiError) {
+				alertStore.showAlert(
+					`${i18n.tr.admin.calendarEventUpdateError}: ${apiError}`,
+					"",
+					true,
+					true,
+				);
+				return;
+			}
+
+			successMessage = i18n.tr.admin.calendarEventUpdatedSuccess;
+			setTimeout(() => {
+				successMessage = null;
+			}, 5000);
+		} else {
+			const createData: CalendarEventCreate = {
+				title: editForm.title,
+				description: editForm.description,
+				external_link: editForm.external_link,
+				event_date: editForm.event_date,
+			};
+
+			const { error: apiError } = await createEvent({
+				body: createData,
+			});
+
+			if (apiError) {
+				alertStore.showAlert(
+					`${i18n.tr.admin.calendarEventCreateError}: ${apiError}`,
+					"",
+					true,
+					true,
+				);
+				return;
+			}
+
+			successMessage = i18n.tr.admin.calendarEventCreatedSuccess;
+			setTimeout(() => {
+				successMessage = null;
+			}, 5000);
+		}
+
+		showEditModal = false;
+		await loadCalendarEvents();
+	} catch (e) {
+		alertStore.showAlert(
+			`${i18n.tr.admin.calendarEventCreateError}: ${e instanceof Error ? e.message : String(e)}`,
+			"",
+			true,
+			true,
+		);
+	}
+}
+
+async function confirmDeleteCalendarEvent() {
+	if (!currentCalendarEvent) return;
+
+	try {
+		const { error: apiError } = await deleteEvent({
+			path: { event_id: currentCalendarEvent.id },
+		});
+
+		if (apiError) {
+			alertStore.showAlert(
+				`${i18n.tr.admin.calendarEventDeleteError}: ${apiError}`,
+				"",
+				true,
+				true,
+			);
+			return;
+		}
+
+		successMessage = i18n.tr.admin.calendarEventDeletedSuccess;
+		setTimeout(() => {
+			successMessage = null;
+		}, 5000);
+		showDeleteModal = false;
+		await loadCalendarEvents();
+	} catch (e) {
+		alertStore.showAlert(
+			`${i18n.tr.admin.calendarEventDeleteError}: ${e instanceof Error ? e.message : String(e)}`,
+			"",
+			true,
+			true,
+		);
+	}
+}
+
+function formatDateGerman(dateString: string): string {
```

Comment:
we're already using [luxon](https://moment.github.io/luxon/#/) elsewhere which offers various parsing / formatting utils - e.g. I think something like this would do something similar: `DateTime(dateString).setLocale('de').toLocaleString()`